src/packs/checker/architecture.rs

```diff
@@ -35,40 +35,43 @@ impl Layers {
     }
 }
 
+impl Checker {
+    fn validate_pack(&self, pack: &Pack) -> Option<String> {
+        match &pack.layer {
+            Some(layer) => {
+                if self.layers.layers.contains(layer) {
+                    None
+                } else {
+                    Some(format!(
+                        "Invalid 'layer' option in '{}'. `layer` must be one of the layers defined in `packwerk.yml`",
+                        &pack.relative_yml().to_string_lossy()
+                    ))
+                }
+            }
+            None => match &pack.enforce_architecture {
+                Some(setting) => {
+                    if setting.is_false() {
+                        None
+                    } else {
+                        Some(format!(
+                                "'layer' must be specified in '{}' because `enforce_architecture` is true or strict.",
```

Comment:
This makes sense... if a pack enforces architecture, it must specify a layer. Surprised we didn't have this before!

Reply:
Huh yeah, I double checked and it definitely just let it fly before!